online-shop/src/app/cart/cart.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, OnInit } from '@angular/core';
+import { CartService } from '../CartService';
+import { IProduct } from '../IProduct';
+import { IProductWrapper } from '../IProductWrapper';
+import { ProductService } from '../service/productService';
+
+@Component({
+  selector: 'app-cart',
+  templateUrl: './cart.component.html',
+  styleUrls: ['./cart.component.css']
+})
+export class CartComponent implements OnInit {
+
+  constructor(private cartService : CartService, private productService: ProductService) { }
+  cart: IProductWrapper[];
+  products: IProduct[] = [];
+  ngOnInit(): void {
+      this.cart = this.cartService.getCart();
+      this.allProductsFromChart();
+  }
+  
+  allProductsFromChart(){
+      this.cart.forEach(element => {
```

Comment:
**suggestion:** you can transform each cart element in an observable (basically what you have a done without the subscribe part), keep that transformation inside a const variable. Then use the [forkJoin](https://www.learnrxjs.io/learn-rxjs/operators/combination/forkjoin) operator and subscribe to the get the whole products result at once.